### ISSUE TITLE
Add vector and control register type

### ISSUE THREAD

**Issue by Rot127** [rizinorg/rizin#1287]
**Is your feature request related to a problem? Please describe.**

The Hexagon architecture has vector registers (for Hexagon HVX) and control registers (hardware loop counts, frame limits etc.). The register types in [reg.h](https://github.com/rizinorg/rizin/blob/dev/librz/include/rz_reg.h) have non of those two.

**Describe the solution you'd like**
Add the vector and control register as type to `reg.h` and `reg.c`.

**Describe alternatives you've considered**
Non really. Maybe reusing the x86 special types?

**Additional context**
I rework the Hexagon disassembler at the moment and have trouble to decide on a register type for the register profile.
If this solution is fine, I will create a pull request.

**Edit**: There are also Guest registers and Predicate registers.
Basically I would like to know, whether a pull request with X many new register types would be accepted.
